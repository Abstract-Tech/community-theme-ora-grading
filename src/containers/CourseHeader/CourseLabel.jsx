import React from 'react';
import PropTypes from 'prop-types';

export const CourseLabel = ({
  courseOrg,
  courseNumber,
  courseTitle,
}) => (
  <div
    className="flex-grow-1 course-title-lockup"
    style={{ lineHeight: 1 }}
  >
    <span className="d-block small m-0">
      {courseOrg} {courseNumber}
    </span>
    <span className="d-block m-0 font-weight-bold course-title">
      {courseTitle}
    </span>
  </div>
);
CourseLabel.propTypes = {
  courseOrg: PropTypes.string,
  courseNumber: PropTypes.string,
  courseTitle: PropTypes.string,
};
CourseLabel.defaultProps = {
  courseOrg: null,
  courseNumber: null,
  courseTitle: null,
};

export default CourseLabel;
