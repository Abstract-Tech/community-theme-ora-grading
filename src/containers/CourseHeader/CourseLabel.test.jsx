import React from 'react';
import { shallow } from 'enzyme';

import CourseLabel from './CourseLabel';

const courseData = {
  courseOrg: 'course-org',
  courseNumber: 'course-number',
  courseTitle: 'course-title',
};

describe('Header CourseLabel component', () => {
  test('snapshot', () => {
    expect(
      shallow(<CourseLabel {...courseData} />),
    ).toMatchSnapshot();
  });
});
