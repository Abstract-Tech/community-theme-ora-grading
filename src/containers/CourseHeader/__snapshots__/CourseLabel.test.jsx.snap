// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header CourseLabel component snapshot 1`] = `
<div
  className="flex-grow-1 course-title-lockup"
  style={
    Object {
      "lineHeight": 1,
    }
  }
>
  <span
    className="d-block small m-0"
  >
    course-org
     
    course-number
  </span>
  <span
    className="d-block m-0 font-weight-bold course-title"
  >
    course-title
  </span>
</div>
`;
